frame/staking/src/default_weights.rs

```diff
@@ -0,0 +1,181 @@
+		(474310000 as Weight)
+			.saturating_add((19672000 as Weight).saturating_mul(s as Weight))
+			.saturating_add(DbWeight::get().reads(7 as Weight))
+			.saturating_add(DbWeight::get().writes(8 as Weight))
+			.saturating_add(DbWeight::get().writes((1 as Weight).saturating_mul(s as Weight)))
+	}
+	fn validate() -> Weight {
+		(99283000 as Weight)
+			.saturating_add(DbWeight::get().reads(2 as Weight))
+			.saturating_add(DbWeight::get().writes(2 as Weight))
+	}
+	fn nominate(n: u32, ) -> Weight {
+		(126479000 as Weight)
+			.saturating_add((5329000 as Weight).saturating_mul(n as Weight))
+			.saturating_add(DbWeight::get().reads(3 as Weight))
+			.saturating_add(DbWeight::get().writes(2 as Weight))
+	}
+	fn chill() -> Weight {
+		(97447000 as Weight)
+			.saturating_add(DbWeight::get().reads(2 as Weight))
+			.saturating_add(DbWeight::get().writes(2 as Weight))
+	}
+	fn set_payee() -> Weight {
+		(74199000 as Weight)
+			.saturating_add(DbWeight::get().reads(1 as Weight))
+			.saturating_add(DbWeight::get().writes(1 as Weight))
+	}
+	fn set_controller() -> Weight {
+		(153911000 as Weight)
+			.saturating_add(DbWeight::get().reads(3 as Weight))
+			.saturating_add(DbWeight::get().writes(3 as Weight))
+	}
+	// WARNING! Some components were not used: ["c"]
+	fn set_validator_count() -> Weight {
+		(10890000 as Weight)
+			.saturating_add(DbWeight::get().writes(1 as Weight))
+	}
+	fn force_no_eras() -> Weight {
+		(13143000 as Weight)
+			.saturating_add(DbWeight::get().writes(1 as Weight))
+	}
+	fn force_new_era() -> Weight {
+		(13112000 as Weight)
+			.saturating_add(DbWeight::get().writes(1 as Weight))
+	}
+	fn force_new_era_always() -> Weight {
+		(13108000 as Weight)
+			.saturating_add(DbWeight::get().writes(1 as Weight))
+	}
+	fn set_invulnerables(v: u32, ) -> Weight {
+		(14140000 as Weight)
+			.saturating_add((218000 as Weight).saturating_mul(v as Weight))
+			.saturating_add(DbWeight::get().writes(1 as Weight))
+	}
+	fn force_unstake(s: u32, ) -> Weight {
+		(344355000 as Weight)
+			.saturating_add((19704000 as Weight).saturating_mul(s as Weight))
+			.saturating_add(DbWeight::get().reads(4 as Weight))
+			.saturating_add(DbWeight::get().writes(8 as Weight))
+			.saturating_add(DbWeight::get().writes((1 as Weight).saturating_mul(s as Weight)))
+	}
+	fn cancel_deferred_slash(s: u32, ) -> Weight {
+		(260009398000 as Weight)
+			.saturating_add((1675144000 as Weight).saturating_mul(s as Weight))
+			.saturating_add(DbWeight::get().reads(1 as Weight))
+			.saturating_add(DbWeight::get().writes(1 as Weight))
+	}
+	fn payout_stakers(n: u32, ) -> Weight {
+		(71905000 as Weight)
+			.saturating_add((274832000 as Weight).saturating_mul(n as Weight))
+			.saturating_add(DbWeight::get().reads(5 as Weight))
+			.saturating_add(DbWeight::get().reads((3 as Weight).saturating_mul(n as Weight)))
+			.saturating_add(DbWeight::get().writes(1 as Weight))
+			.saturating_add(DbWeight::get().writes((1 as Weight).saturating_mul(n as Weight)))
+	}
```

Comment:
this one has very changed from previously:
```rust
/// DB Weight:
/// - Read: EraElectionStatus, CurrentEra, HistoryDepth, ErasValidatorReward,
///         ErasStakersClipped, ErasRewardPoints, ErasValidatorPrefs (8 items)
/// - Read Each: Bonded, Ledger, Payee, Locks, System Account (5 items)
/// - Write Each: System Account, Locks, Ledger (3 items)
/// # </weight>
#[weight =
	120 * WEIGHT_PER_MICROS
	+ 54 * WEIGHT_PER_MICROS * Weight::from(T::MaxNominatorRewardedPerValidator::get())
	+ T::DbWeight::get().reads(7)
	+ T::DbWeight::get().reads(5)  * Weight::from(T::MaxNominatorRewardedPerValidator::get() + 1)
	+ T::DbWeight::get().writes(3) * Weight::from(T::MaxNominatorRewardedPerValidator::get() + 1)
]
```
it seems having one write per nominator is wrong considering Ledge, locks and account data should be written for each of them.
and same for the read which as far as I understand should also be 5.

also it seems the reads/writes for the validator are also not taking into account

Reply:
FYI I ran this again locally and I get the following. Strange that the read/write analysis is so different: 

```
========
-- Extrinsic Time --

Model:
Time ~=        0
    + n    305.5
              µs

Reads = 12 + (3 * n)
Writes = 3 + (1 * n)
Min Squares Analysis
========
-- Extrinsic Time --

Data points distribution:
    n   mean µs  sigma µs       %
    1     499.8     16.19    3.2%
    7     482.1     5.734    1.1%
   13      3613     73.67    2.0%
   19      4990     187.6    3.7%
   25      7648     207.4    2.7%
   31      9702     203.5    2.0%
   37     11330     36.63    0.3%
   43     13140     66.04    0.5%
   49     14720     105.8    0.7%
   55     16490     61.93    0.3%
   61     18150     163.8    0.9%

Quality and confidence:
param     error
n         2.646

Model:
Time ~=        0
    + n    311.6
              µs

Reads = 5 + (3 * n)
Writes = 1 + (1 * n)
```

Reply:
It must be because of some issue with the benchmark I reckon. Maybe we are not paying the controller, but rather the stash and thus less db access.

Reply:
The comments say: 

```
///   NOTE: weights are assuming that payouts are made to dead controller accounts.
///   Currently we don't refund otherwise.
```

But the database counts noted in the previous round are for the case that we pay rewards to a `Staked` entity (i.e. writing to ledger is only for staked). So Everything is correct, it is just a matter of which variant we should use as the worse-case of this function?

1. Reward all to dead Controller? (current)
2. Reward all to alive Staked?

Reply:
yes so I made a new benchmark that makes all payouts to alive staked accounts, and we get: 

```
Pallet: "pallet_staking", Extrinsic: "payout_stakers_alive_staked", Lowest values: [], Highest values: [], Steps: [10], Repeat: 20
Median Slopes Analysis
========
-- Extrinsic Time --

Model:
Time ~=    49.83
    + n    55.16
              µs

Reads = 12 + (5 * n)
Writes = 3 + (3 * n)
Min Squares Analysis
```

Reply:
@kianenigma you should have a benchmark for both situations and do `payout_staked.max(payout_controller_new)`

Reply:
I don't there's a need for `max`ing at runtime, `payout_stakers_alive_staked ` is clearly more expensive based on the outputs above.